Mantis 45314: Access to roleassignment fails for position-holders

ingoj opened the pull request:
Updated the permission-check in roleassignmentObject as a possible fix for the mantis-issue.
Languages de/en are adjusted, too, since the error message which was given was sometines misleading.

ingoj committed:
Fix access check for read_users

Access to roleassignments-tab still checked read_users when using position permissions. I changed to match then checkWriteRights- functon in this class.

ingoj committed:
Adjust error message when access to roleassingnment fails

This message may appear if users may read the roleassignments, too, even if they don't want to change anything.
Gender-switched from "Benutzer" to "Konto"

ingoj committed:
clarify message when accessing role assignments od a user

This message may appear if users want to view roleassignments, even if they don't want to change anything

kergomard commented:
Thank you for the PR @ingoj !

I've just one small gripe with it: We shouldn't change the content of the language variable, as it is used in more than one place and sometimes it fits as it is. Could you please introduce a second one "msg_no_perm_view_roles_of_user" and add your texts there.

Thanks again and best,
@kergomard

ingoj commented:
I will send a new PR soon. The new PR will have the changes to the langvar and adittionally fix the problem with the filter.

ingoj committed:
Merge branch 'ILIAS-eLearning:release_9' into release_9_roles_positions

ingoj committed:
Nowe error message if access to roleassignments fail

additionlly switches 2 entries from "Benutzer" to "Konto"

ingoj committed:
New error message if access to roleassignments fail

ingoj committed:
use new error message

ingoj committed:
Do not show dysfunctional filter for users with position access

I set the filter value to 0 for users that don't have edit access.
For this I introduces da new varaible in the constructor, since this will help to fix another issue (role that cant be edited are clickable and lead to an error message) in the future.
I used this variable in the constructor, too.
Since rbacreview doesn't offer a constant for FILTER_ASSIGNED I used the fixed value like it is done in the rbacreview class

ingoj committed:
remove checkbox-column if edit_access is not granted

and fix order of statements for variable definition (edit_access) in constructor

ingoj commented:
I updated the PR according to the discussion
Additionally I removed the dysfunctional filter in the roleassignments table

kergomard commented:
Thank you very much @ingoj !

We have a few more issues and then we are done:
- You have to add the property `ilRoleAssignmentTableGUI::edit_access` explicitly otherwise php will complain.
- Please run the cs-fixer on your changed files as there are errors (See: https://github.com/ILIAS-eLearning/ILIAS/blob/release_9/docs/development/php-coding-style.md#code-style-checks-and-fixes).

I think we can then run with this.

Thanks again and best,
@kergomard

ingoj committed:
apply cs fixes

ingoj committed:
declare new variable and cs fixes

ingoj committed:
fix typo

ingoj commented:
I ran cs-fixer and declared the variable.

@kergomard  Should be o.k. now.

kergomard commented:
Hi @ingoj 
As now after the cs-fixer complained also the lang-order was wrong, I decided to clean this up myself. You are still credited as the author of the corresponding commits. I additionally removed the tab completely, if somebody doesn't have read access to it.

See:
- https://github.com/ILIAS-eLearning/ILIAS/commit/a66f1499f076716279f3716901e821a5a9670804
- https://github.com/ILIAS-eLearning/ILIAS/commit/3df2d5c0e10422e3aaf70bc5b8155be61e45f73f
- https://github.com/ILIAS-eLearning/ILIAS/commit/cd7f4068f207c7eb28a0d37ff3bc6f2cb67e5a31
- https://github.com/ILIAS-eLearning/ILIAS/commit/278d443e254d40394e3ae74f5f1d28d7d56ad523
- https://github.com/ILIAS-eLearning/ILIAS/commit/4222d44821ba73f1afe8a785892a8637dda82219
- https://github.com/ILIAS-eLearning/ILIAS/commit/7416d7588477270bd7f1f5249c446e5d8a665590

Everything picked to ILIAS 10 and trunk.

Best,
@kergomard
